test/integration/world-component.spec.ts

```diff
@@ -0,0 +1,262 @@
+import { createWorldsComponent } from '../../src/adapters/worlds'
+import { PermissionType } from '../../src/types'
+import { InvalidRequestError } from '../../src/types/errors'
+
+describe('WorldComponent', () => {
+  const mockFetch = jest.fn()
+  let worldsComponent: Awaited<ReturnType<typeof createWorldsComponent>>
+
+  beforeEach(async () => {
+    jest.clearAllMocks()
+
+    const mockConfig = {
+      requireString: jest.fn().mockImplementation((key) => {
+        const values = {
+          WORLD_CONTENT_URL: 'https://world-content.test',
+          LAMBDAS_URL: 'https://lambdas.test/'
+        }
+        return Promise.resolve(values[key] || '')
+      }),
+      getString: jest.fn(),
+      getNumber: jest.fn(),
+      requireNumber: jest.fn()
+    }
+
+    const mockLogs = {
+      getLogger: jest.fn().mockReturnValue({
+        log: jest.fn(),
+        warn: jest.fn(),
+        info: jest.fn()
+      })
+    }
+
+    const mockCachedFetch = {
+      cache: jest.fn().mockImplementation(() => ({
+        fetch: mockFetch
+      }))
+    }
+
+    worldsComponent = await createWorldsComponent({
+      config: mockConfig,
+      cachedFetch: mockCachedFetch,
+      logs: mockLogs
+    })
+  })
+
+  describe('hasWorldOwnerPermission', () => {
+    beforeEach(() => {
+      const mockNamesResponse = {
+        elements: [{ name: 'myworld' }, { name: 'otherworld' }]
+      }
+
+      mockFetch.mockImplementation(() => Promise.resolve(mockNamesResponse))
+    })
+
+    it('should throw an error when no world name is provided', async () => {
+      await expect(worldsComponent.hasWorldOwnerPermission('authAddress', '')).rejects.toThrow(
+        'Invalid world name: , should end with .dcl.eth or .eth'
+      )
+
+      await expect(worldsComponent.hasWorldOwnerPermission('authAddress', '')).rejects.toBeInstanceOf(
+        InvalidRequestError
+      )
+    })
+
+    it("should throw an error when world name doesn't end with .eth", async () => {
+      await expect(worldsComponent.hasWorldOwnerPermission('authAddress', 'myworld')).rejects.toThrow(
+        'Invalid world name: myworld, should end with .dcl.eth or .eth'
+      )
+
+      await expect(worldsComponent.hasWorldOwnerPermission('authAddress', 'myworld')).rejects.toBeInstanceOf(
+        InvalidRequestError
+      )
```

Comment:
This test can be simplified using:
```suggestion
      await expect(worldsComponent.hasWorldOwnerPermission('authAddress', 'myworld')).rejects.toThrow(
        'Invalid world name: myworld, should end with .dcl.eth or .eth'
      )

      await expect(worldsComponent.hasWorldOwnerPermission('authAddress', 'myworld')).rejects.toThrow(
        new InvalidRequestError('Invalid world name: myworld, should end with .dcl.eth or .eth')
      )
```
The same applies to other tests here.